### DISCUSSION TITLE
Renovate remove dependencies in package-lock.json

### DISCUSSION THREAD

**Discussion by HeadFox** [renovatebot/renovate#13420] [Request Help]
Hello :)
We are self-hosting renovate using the renovate/renovate image and binarySource=docker. 
We are experiencing some issues with the package-lock.json v2 update.
Here are the dependencies updated : 
![image](https://user-images.githubusercontent.com/6277284/148531182-593ff4a4-275e-4cc1-85a6-33d6d1d943f3.png)
The problem is that if I rerun an `npm install` locally on the renovate branch. The package-lock.json does not correspond....
For exemple the line with jest@27.4.6 is removed but the new jest@27.4.7 is not added
![image](https://user-images.githubusercontent.com/6277284/148531526-faecd93d-cc17-4f60-8e08-8eee07031047.png)

But locally if I rerun npm install :
![image](https://user-images.githubusercontent.com/6277284/148531634-d944ad0b-3662-41b5-9b0b-f02efe10af9c.png)
jest@27.4.7 is correctly added to the lock file.

Maybe I am missing some specific option in renovate config

**Comment by rarkins** [comment]
If you can reproduce the same problem in a public GitHub repo then we can take a look

**Reply by HeadFox** [reply]
Finally we found that the problem was coming from permissions problem in our pipeline.
We decided to use the `binarySource: install` since it's a lot easier to make it work :)
Note: if you someone is using Jenkins you need to set the BUILDPACK en variable manually to make it work (even with the use of `renovate/renovate` docker image
```
 environment {
        BUILDPACK = "1"
    }
```

**Reply by viceice** [reply]
It's because Jenkins is using `docker exec` to run commands, which won't load our env.
